eventstore/views.py

```diff
@@ -20,6 +27,67 @@
     PrebirthRegistrationSerializer,
     PublicRegistrationSerializer,
 )
+from ndoh_hub.utils import TokenAuthQueryString, validate_signature
+
+
+class MessagesViewSet(GenericViewSet):
+    queryset = Message.objects.all()
+    permission_classes = (DjangoModelPermissions,)
+    authentication_classes = (TokenAuthQueryString, TokenAuthentication)
+
+    def create(self, request):
+        validate_signature(request)
+        webhook_type = request.headers.get("X-Turn-Hook-Subscription", None)
+        if webhook_type == "whatsapp":
+            for inbound in request.data.get("messages", []):
+                id = inbound.pop("id")
+                contact_id = inbound.pop("from")
+                type = inbound.pop("type")
+                timestamp = datetime.fromtimestamp(int(inbound.pop("timestamp")))
+
+                Message.objects.create(
+                    id=id,
+                    contact_id=contact_id,
+                    type=type,
+                    data=inbound,
+                    message_direction=Message.INBOUND,
+                    created_by=request.user.username,
+                    timestamp=timestamp,
+                )
+
+            for statuses in request.data.get("statuses", []):
+                message_id = statuses.pop("id")
+                recipient_id = statuses.pop("recipient_id")
+                timestamp = datetime.fromtimestamp(int(statuses.pop("timestamp")))
+                message_status = statuses.pop("status")
+                Event.objects.create(
+                    message_id=message_id,
+                    recipient_id=recipient_id,
+                    timestamp=timestamp,
+                    status=message_status,
+                    created_by=request.user.username,
+                    data=statuses,
+                )
+
+        elif webhook_type == "turn":
+            outbound = request.data
+            data = {
+                "text": outbound["text"],
+                "render_mentions": outbound["render_mentions"],
+                "preview_url": outbound["preview_url"],
+            }
+            Message.objects.create(
+                id=request.headers["X-WhatsApp-Id"],
+                contact_id=outbound["to"],
+                type=outbound["type"],
+                data=data,
+                message_direction=Message.OUTBOUND,
+                created_by=request.user.username,
+                recipient_type=outbound["recipient_type"],
+            )
+        else:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
We should probably explain here to the user what they did wrong

Reply:
what is the best way to explain to the user what went wrong?